### PR TITLE
[JUJU-3288] Add image-id constraint

### PR DESCRIPTION
**Pull request by nvinuesa**
This PR adds the new image-id constraint that has been added to juju core in https://github.com/juju/juju/pull/15284.

QA:
```sh
$ go test ./...
ok  	github.com/juju/description/v4	0.469s
```
also, `$ make check-go` passes.

### CODE REVIEW

File: constraints_test.go [hunk 1]
```diff
@@ -292,3 +299,48 @@ func (s *ConstraintsSerializationSuite) TestParsingV4Minimal(c *gc.C) {
 	expected := &constraints{Version: 4}
 	c.Assert(imported, gc.DeepEquals, expected)
 }
+
+//
```
Comment (SimonRichardson) [inline constraints_test.go:303]: Delete me

**Review by SimonRichardson — CHANGES_REQUESTED** [review]

File: cloudimagemetadata.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ type cloudimagemetadata struct {
 	DateCreated_     int64      `yaml:"date-created"`
 	Source_          string     `yaml:"source"`
 	Priority_        int        `yaml:"priority"`
-	ImageId_         string     `yaml:"image-id"`
+	ImageID_         string     `yaml:"image-id"`
```
[the remark this answers is not in the document]
Comment (nvinuesa) [reply cloudimagemetadata.go]: Oh you are absolutely right, it's reverted now

File: constraints_test.go (hunk 1 above)
Comment (nvinuesa) [reply constraints_test.go]: done

File: interfaces.go [hunk 3]
```diff
@@ -78,7 +78,7 @@ type CloudImageMetadata interface {
 	DateCreated() int64
 	Source() string
 	Priority() int
-	ImageId() string
+	ImageID() string
```
[the remark this answers is not in the document]
Comment (nvinuesa) [reply interfaces.go]: done

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**